integration-tests/applications/src/test/java/com/xsk/integration/tests/applications/deployment/client/XSKHttpClient.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.xsk.integration.tests.applications.deployment.client;
+
+
+import com.xsk.integration.tests.applications.Deployment;
+import com.xsk.integration.tests.applications.deployment.DeploymentException;
+import com.xsk.integration.tests.applications.kyma.TokenProvider;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.concurrent.FutureCallback;
+import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
+import org.apache.http.impl.nio.client.HttpAsyncClients;
+import org.apache.http.message.BasicHeader;
+
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+
+import static com.xsk.integration.tests.applications.Deployment.KYMA;
+import static com.xsk.integration.tests.applications.Deployment.LOCAL;
+
+
+public class XSKHttpClient {
+
+    private CloseableHttpAsyncClient httpClient;
+
+    public XSKHttpClient(Deployment deployment) {
+        if (deployment == KYMA) {
+            TokenProvider tokenProvider = new TokenProvider();
+            String kymaToken = tokenProvider.getToken();
+            BasicHeader authHeader = new BasicHeader(HttpHeaders.AUTHORIZATION, "Bearer " + kymaToken);
+            httpClient = HttpAsyncClients.custom().setDefaultHeaders(List.of(authHeader)).build();
+        } else if (deployment == LOCAL) {
+            httpClient = HttpAsyncClients.createDefault();
+        }
+        httpClient.start();
+    }
+
+
+    CompletableFuture<HttpResponse> executeRequestAsync(HttpUriRequest request) {
+        var future = new CompletableHttpResponseFuture();
+        httpClient.execute(request, future);
+        return future;
+    }
+
+
+    static class CompletableHttpResponseFuture extends CompletableFuture<HttpResponse> implements FutureCallback<HttpResponse> {
+
+        @Override
+        public void completed(HttpResponse httpResponse) {
+            int statusCode = httpResponse.getStatusLine().getStatusCode();
+            if (statusCode > 299 || statusCode < 200) {
```

Comment:
Maybe we should think of a better way to handle errors from the APIs? We can at least make `CompletableHttpResponseFuture` accept in its constructor the API that was called and add it to the `DeploymentExceptionMessage`